datajoint/expression.py

```diff
@@ -119,17 +121,37 @@ def where_clause(self):
             else " WHERE (%s)" % ")AND(".join(str(s) for s in self.restriction)
         )
 
+    def sorting_clauses(self):
+        if not self._top:
+            return ""
+        limit = self._top.limit
+        order_by = self._top.order_by
+        offset = self._top.offset
+
+        clause = (
+            ""
+            if not self.primary_key and order_by == ["KEY"]
```

Comment:
Does this logic work for `["KEY desc"]`, or `["KEY "]`?

Reply:
perhaps this logic could be rolled into `_flatten_attribute_list`?

Reply:
It wouldn't, good catch. I'll convert it to regex matching.